builtin/mainmenu/dlg_config_world.lua

```diff
@@ -46,60 +42,49 @@ local function get_formspec(data)
 		"label[0,3.875;" .. fgettext("Optional dependencies:") .. "]" ..
 		"textlist[0,4.375;5,1.8;world_config_optdepends;" ..
 		soft_deps .. ";0]" ..
-		"button[3.25,7;2.5,0.5;btn_config_world_save;" .. fgettext("Save") .. "]" ..
-		"button[5.75,7;2.5,0.5;btn_config_world_cancel;" .. fgettext("Cancel") .. "]"
+		"button[3.25,7;2.5,0.5;btn_config_world_save;" .. fgettext("Save") ..
+		"]" ..
+		"button[5.75,7;2.5,0.5;btn_config_world_cancel;" ..
+		fgettext("Cancel") .. "]"
 
-	if mod and mod.name ~= "" and mod.typ ~= "game_mod" then
+	if mod.name ~= "" and mod.typ ~= "game_mod" then
 		if mod.is_modpack then
-			local rawlist = data.list:get_raw_list()
-
-			local all_enabled = true
-			for j = 1, #rawlist, 1 do
-				if rawlist[j].modpack == mod.name and not rawlist[j].enabled then
-					all_enabled = false
-					break
-				end
-			end
 
-			if all_enabled then
+			if modmgr.is_modpack_entirely_enabled(data, mod.name) then
 				retval = retval .. "button[5.5,0.125;2.5,0.5;btn_mp_disable;" ..
 					fgettext("Disable MP") .. "]"
 			else
 				retval = retval .. "button[5.5,0.125;2.5,0.5;btn_mp_enable;" ..
 					fgettext("Enable MP") .. "]"
 			end
 		else
-			if mod.enabled then
-				retval = retval .. "checkbox[5.5,-0.125;cb_mod_enable;" ..
-					fgettext("enabled") .. ";true]"
-			else
-				retval = retval .. "checkbox[5.5,-0.125;cb_mod_enable;" ..
-					fgettext("enabled") .. ";false]"
-			end
+			retval = retval .. "checkbox[5.5,-0.125;cb_mod_enable;" ..
+				fgettext("enabled") .. ";" .. tostring(mod.enabled) .. "]"
 		end
 	end
 	if enabled_all then
 		retval = retval ..
-			"button[8.75,0.125;2.5,0.5;btn_disable_all_mods;" .. fgettext("Disable all") .. "]"
+			"button[8.75,0.125;2.5,0.5;btn_disable_all_mods;" ..
+			fgettext("Disable all") .. "]"
 	else
 		retval = retval ..
-			"button[8.75,0.125;2.5,0.5;btn_enable_all_mods;" .. fgettext("Enable all") .. "]"
+			"button[8.75,0.125;2.5,0.5;btn_enable_all_mods;" ..
+			fgettext("Enable all") .. "]"
 	end
-	retval = retval ..
+	return retval ..
 		"tablecolumns[color;tree;text]" ..
-		"table[5.5,0.75;5.75,6;world_config_modlist;"
-	retval = retval .. modmgr.render_modlist(data.list)
-	retval = retval .. ";" .. data.selected_mod .."]"
-
-	return retval
+		"table[5.5,0.75;5.75,6;world_config_modlist;" ..
+		modmgr.render_modlist(data.list) ..
+		";" .. data.selected_mod .."]"
 end
 
 local function enable_mod(this, toset)
 	local mod = this.data.list:get_list()[this.data.selected_mod]
 
-	if mod.typ == "game_mod" then
+	--~ if mod.typ == "game_mod" then
 		-- game mods can't be enabled or disabled
-	elseif not mod.is_modpack then
+	--~ elseif not mod.is_modpack then
+	if not mod.is_modpack then
```

Comment:
This changes the behaviour of game mods. Is it possible to disable/re-enable them with the current mod handling?